geo/src/algorithm/map_coords.rs

```diff
@@ -165,6 +165,45 @@ pub trait MapCoordsInplace<T> {
         T: CoordNum;
 }
 
+/// Map a fallible function over all the coordinates in a geometry, returning a Result
+pub trait TryMapCoordsInplace<T, E> {
+    /// Map a fallible function over all the coordinates in a geometry, in place, returning a `Result`.
+    ///
+    /// Upon encountering an `Err` from the function, `try_map_coords_inplace` immediately returns
+    /// and the geometry is potentially left in a partially mapped state.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo::algorithm::map_coords::TryMapCoordsInplace;
+    ///
+    /// let mut p1 = geo::point!{x: 10u32, y: 20u32};
+    ///
+    /// p1.try_map_coords_inplace(|&(x, y)| -> Result<_, &str> {
+    ///     Ok((
+    ///         x.checked_add(1000).ok_or("Overflow")?,
+    ///         y.checked_mul(2).ok_or("Overflow")?,
+    ///     ))
+    /// })?;
+    ///
+    /// assert_eq!(
+    ///     p1,
+    ///     geo::point!{x: 1010u32, y: 40u32},
+    /// );
+    /// # Ok::<(), &str>(())
+    /// ```
+    fn try_map_coords_inplace(
```

Comment:
```suggestion
    fn try_map_coords_in_place(
```

I think "In place" should be spelled as two words, like: https://doc.rust-lang.org/std/ptr/fn.drop_in_place.html?search=in_place (7 matches)

Reply:
Wild addendum: We have a pattern in this module of "one method per trait": TryMapCoordsInplace, MapCoords, TryMapCoords, MapCoordsInplace

I think it might be more convenient for users to have all these grouped under a single MapCoords trait. Is there any reason you can think of where it'd be good to keep them separate? What would you think about, instead of introducing this new `TryMapCoordsInplace` trait, instead deprecating `TryMapCoords` and `MapCoordsInplace` and consolidating all their methods (with corrected spelling) on the `MapCoords` trait?
 
Because you're following the existing pattern, I think it's reasonable to merge as is, so consider this non-blocking feedback.

If you'd prefer not to make these changes in this PR, I'll probably follow up with my own PR proposing something like what I outlined above.

Reply:
Regarding the naming, I noticed that too when I was writing this pull request. I named it this way to follow existing convention, but agree we should rename it. I opened this GitHub Issue to track that: https://github.com/georust/geo/issues/802

And regarding merging them into a single trait, I completely agree. There's already a separate issue for that: https://github.com/georust/geo/issues/721